src/Microsoft.OData.Core/UriParser/SemanticAst/ODataPathExtensions.cs

```diff
@@ -55,6 +55,19 @@ public static bool IsCollection(this ODataPath path)
             return path.LastSegment.TranslateWith(new IsCollectionTranslator());
         }
 
+        /// <summary>
+        /// Creates a new ODataPath with the specified segment appended
+        /// </summary>
+        /// <param name="path">The path against which the segment should be applied.</param>
+        /// <param name="segment">The segment applied to the path.</param>
+        /// <returns>A new ODataPath with the segment appended.</returns>
+        public static ODataPath AppendSegment(this ODataPath path, ODataPathSegment segment)
+        {
+            var newPath = new ODataPath(path);

```

Comment:
Append* sounds to insert the item at the end and to return the same "path"?

Reply:
Yes' Append does sound like it just adds to the end, but this naming is consistent with AppendEntityKeySegment, AppendNavigationPropertySegment, AppendPropertySegment, and AppendKeySegment, so I kept the naming consistent, rather than changing it everywhere.

---
In reply to: [532791397](https://github.com/OData/odata.net/pull/1930#discussion_r532791397) [](ancestors = 532791397)